src/codegen.cpp

```diff
@@ -2688,12 +2707,37 @@ static bool emit_builtin_call(jl_codectx_t &ctx, jl_cgval_t *ret, jl_value_t *f,
                             data_owner->addIncoming(aryv, curBB);
                             data_owner->addIncoming(own_ptr, ownedBB);
                         }
-                        typed_store(ctx,
-                                    emit_arrayptr(ctx, ary, args[1], isboxed),
-                                    idx, v, ety,
-                                    !isboxed ? tbaa_arraybuf : tbaa_ptrarraybuf,
-                                    data_owner, 0,
-                                    false); // don't need to root the box if we had to make one since it's being stored in the array immediatly
+                        if (jl_is_uniontype(ety)) {
+                            Value *nbytes = ConstantInt::get(T_size, elsz);
+                            Value *data = emit_arrayptr(ctx, ary, args[1]);
+                            if (!(rhs.typ == jl_bottom_type)) {
```

Comment:
Not sure if this is the right kind of check here; the issue is when you have something like
```julia
v = Vector{Union{Int8, Void}}(10);
v[1] = Int64(10)
```
this should result in a `convert` error, but it passes thru codegen with `rhs` being `jl_bottom_type`, which ends up throwing an error in `compute_tindex_unboxed` (since Bottom type doesn't match one of the Union types).